src/debug-tools/components/stores-tree.tsx

```diff
@@ -42,98 +35,88 @@ export const columns: IColumn[] = [
     },
 ];
 
-export class StoresTree extends React.Component<StoresTreeProps, StoresTreeState> {
-    private selection = new Selection();
-    private selectionMode = SelectionMode.none;
-    private compact = true;
-
-    constructor(props: StoresTreeProps) {
-        super(props);
-        this.state = {
-            global: {},
-        };
-    }
-
-    public componentDidMount(): void {
-        this.props.global.forEach(store => {
-            store.addChangedListener(() => {
-                this.setState({
-                    global: {
-                        ...this.state.global,
-                        [store.getId()]: store.getState(),
-                    },
-                });
-            });
-        });
+export interface StoresTreeState {
+    featureFlagStoreData: FeatureFlagStoreData;
+    scopingStoreData: ScopingStoreData;
+    userConfigurationStoreData: UserConfigurationStoreData;
+    permissionsStateStoreData: PermissionsStateStoreData;
+}
 
-        this.props.storeActionMessageCreator.getAllStates();
-    }
+export type StoresTreeDeps = {
+    storesHub: ClientStoresHub<StoresTreeState>;
+};
 
-    public render(): JSX.Element {
-        const global = this.state.global;
+export interface StoresTreeProps {
+    deps: StoresTreeDeps;
+    state: StoresTreeState;
+}
 
-        if (isEmpty(global)) {
-            return <Spinner label="loading..." />;
-        }
+export const StoresTree = NamedFC<StoresTreeProps>('StoresTree', ({ deps, state }) => {
+    const { storesHub } = deps;
 
-        const groups = [];
-        let items = [];
+    if (!storesHub.hasStoreData()) {
+        return <Spinner label="loading..." />;
+    }
 
-        let instanceCount: number = 0;
+    const compact = true;
+    const selection = new Selection();
+    const selectionMode = SelectionMode.none;
 
-        forEach(global, (storeState, storeKey) => {
-            const stateKeys = Object.keys(storeState);
+    const groups = [];
```

Comment:
These probably should have types, no?

Reply:
So... for `groups` we can use `IGroup` and I updated the PR with that.
But for `items`... office fabric defines items as `any[]` which makes sense but then again, not sure if it makes sense to add a type for `items` as this const is local and private.